packages/api-v4/src/cloudpulse/types.ts

```diff
@@ -412,3 +412,22 @@ export interface CloudPulseAlertsPayload {
    */
   user_alerts?: number[];
 }
+
+export interface CreateNotificationChannelPayload {
+  /**
+   * The type of channel to create.
+   */
+  channel_type: ChannelType;
+  /**
+   * The details of the channel to create.
+   */
+  details: {
```

Comment:
@ankita-akamai ,If possible and not much effort,  we make this more future-proof and extensible. 

We can have a possible type for each of different ChannelType details 

```ts
interface EmailDetail {
  email: {
    usernames: string[];
  }
}
```

Call this 
```suggestion
  details: EmailDetail;
```

Reply:
@santoshp210-akamai  +1